File: frontend/src/api/documents.ts
import { apiRequest } from "./client";
import type { Language } from "../i18n";

export type DocumentType =
  | "davo_arizasi"
  | "shikoyat"
  | "ishonchnoma"
  | "mehnat_shartnomasi"
  | "ijara_shartnomasi"
  | "sud_arizasi";

export type DocumentField = {
  name: string;
  label: string;
  field_type: "text" | "textarea" | "date" | "select";
  required: boolean;
  placeholder: string | null;
  help_text: string | null;
  options: string[];
};

export type DocumentTemplate = {
  type: DocumentType;
  title: string;
  description: string;
  fields: DocumentField[];
};

export type DocumentSource = {
  id: string;
  source_url: string;
  law_title: string;
  article_number: string | null;
  heading: string | null;
};

export type GeneratedDocument = {
  id: string;
  type: DocumentType;
  title: string;
  content: string;
  disclaimer: string;
  sources: DocumentSource[];
  pdf_url: string;
};

export function getDocumentTemplates(language: Language): Promise<DocumentTemplate[]> {
  return apiRequest<DocumentTemplate[]>(`/documents/templates?language=${encodeURIComponent(language)}`);
}

export function generateDocument(
  type: DocumentType,
  values: Record<string, string>,
  language: Language,
): Promise<GeneratedDocument> {
  return apiRequest<GeneratedDocument>("/documents/generate", {
    method: "POST",
    body: JSON.stringify({ type, values, language }),
  });
}

export { resolveApiUrl } from "./client";

File: backend/app/schemas/documents.py
from __future__ import annotations

from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field

from app.models import DocumentType
from app.schemas.mentor import ResponseLanguage


DocumentFieldType = Literal["text", "textarea", "date", "select"]


class DocumentFieldSchema(BaseModel):
    name: str
    label: str
    field_type: DocumentFieldType
    required: bool
    placeholder: str | None = None
    help_text: str | None = None
    options: list[str] = Field(default_factory=list)


class DocumentTemplateSchema(BaseModel):
    type: DocumentType
    title: str
    description: str
    fields: list[DocumentFieldSchema]


class DocumentGenerateRequest(BaseModel):
    type: DocumentType
    values: dict[str, str] = Field(default_factory=dict)
    language: ResponseLanguage = ResponseLanguage.uz_latin


class LegalSourceSchema(BaseModel):
    id: str
    source_url: str
    law_title: str
    article_number: str | None
    heading: str | None


class GeneratedDocumentSchema(BaseModel):
    id: UUID
    type: DocumentType
    title: str
    content: str
    disclaimer: str
    sources: list[LegalSourceSchema]
    pdf_url: str
